enterprise/app/jobs/captain/conversation/response_builder_job.rb

```diff
@@ -3,8 +3,12 @@ class Captain::Conversation::ResponseBuilderJob < ApplicationJob
 
   def perform(conversation, assistant)
     @conversation = conversation
+    @inbox = conversation.inbox
     @assistant = assistant
 
+    # this ensures that the conversation is not processed if the assistant limits are reached
+    return ActiveRecord::Base.transaction { process_action('handoff') } unless @inbox.captain_active?
+
```

Comment:
I think a better approach is to avoid enqueue in the job at all. You update the check on Enterprise::MessageTemplates::HookExecutionService from `inbox.captain_assistant.present?` to `inbox.active_bot?`